### PR TITLE
KEYCLOAK-7888 Better way of securing CXF deployments

### PR DESCRIPTION
**Pull request by hmlnarik**
Documentation for keycloak/keycloak#5399

### CODE REVIEW

File: securing_apps/topics/oidc/java/fuse7/cxf-builtin.adoc [hunk 1]
```diff
@@ -1,10 +1,10 @@
 
 [[_fuse7_adapter_cxf_builtin]]
-===== Securing an Apache CXF Endpoint on the Default Jetty Engine
+===== Securing an Apache CXF Endpoint on the Default Undertow Engine
 
-Some services automatically come with deployed servlets on startup. One such service is the CXF servlet running in the $$http://localhost:8181/cxf$$ context. Securing such endpoints can be complicated. One approach, which {project_name} is currently using, is `ServletReregistrationService` which undeploys a built-in servlet at startup, enabling you to redeploy it on a context secured by {project_name}.
+Some services automatically come with deployed servlets on startup. One such service is the CXF servlet running in the $$http://localhost:8181/cxf$$ context. Fuse's Pax Web supports context configuration via Configuration Admin which can be used to secure endpoints by {project_name}.
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7/cxf-builtin.adoc]: _context configuration_ is not precise - it's rather _altering_/_processing_ of existing contexts. You can refer to https://ops4j1.jira.com/wiki/spaces/paxweb/pages/354025473/HTTP+Context+processing

Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7/cxf-builtin.adoc]: _context configuration_ is not precise - it's rather _altering_/_processing_ of existing contexts. You can refer to https://ops4j1.jira.com/wiki/spaces/paxweb/pages/354025473/HTTP+Context+processing

File: securing_apps/topics/oidc/java/fuse7/cxf-builtin.adoc [hunk 2]
```diff
@@ -28,47 +27,41 @@ The configuration file `OSGI-INF/blueprint/blueprint.xml` inside your applicatio
...
+`security._anyName_.url` and `security._anyName_.roles`::
+  Values of properties of individual security constraints just as they would be set in `security-constraint/web-resource-collection/url-pattern` and `security-constraint/auth-constraint/role-name` in `web.xml`, respectively. Roles are separated by comma and whitespace around it. The `_anyName_` identifier can be arbitrary but must match for individual properties of the same security constraint.
++
+[NOTE]
+====
+Some Fuse versions contain a bug that requires roles to be separated by `", "` (comma and single space). Make sure you use precisely this notation for separating the roles.
+====
 
-* The `Import-Package` in `META-INF/MANIFEST.MF` must contain at least these imports:
+The `Import-Package` in `META-INF/MANIFEST.MF` must contain at least these imports:
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7/cxf-builtin.adoc:64]: In fact, I didn't have to add any explicit imports for `org.keycloak` packages of my bundle containing blueprint-based CXF endpoints. pax-web's processing is fully external to your bundle (and to CXF).

**Review by grgrzybek — CHANGES_REQUESTED** [review]

**Review by matthewhelmke — APPROVED** [review]
Looks good to me.

**Pull request merged**